plugins/nf-tower/src/main/io/seqera/tower/plugin/cli/AuthCommandImpl.groovy

```diff
@@ -0,0 +1,1354 @@
+    }
+
+    private Map pollForDeviceToken(String deviceCode, int intervalSeconds, Map auth0Config) {
+        final tokenUrl = "https://${auth0Config.domain}/oauth/token"
+        def retryCount = 0
+
+        while( retryCount < AUTH_POLL_TIMEOUT_RETRIES ) {
+            // Check for interrupt
+            if( Thread.currentThread().isInterrupted() ) {
+                throw new InterruptedException("Authentication polling interrupted")
+            }
+
+            final params = [
+                'grant_type' : 'urn:ietf:params:oauth:grant-type:device_code',
+                'device_code': deviceCode,
+                'client_id'  : auth0Config.clientId
+            ]
+
+            try {
+                final result = performAuth0Request(tokenUrl, params)
+                return result
+            } catch( RuntimeException e ) {
+                final message = e.message
+                if( message.contains('authorization_pending') ) {
+                    print "${ColorUtil.colorize('.', 'dim', true)}"
+                    System.out.flush()
+                } else if( message.contains('slow_down') ) {
+                    intervalSeconds += 5
+                    print "${ColorUtil.colorize('.', 'dim', true)}"
+                    System.out.flush()
+                } else if( message.contains('expired_token') ) {
+                    throw new RuntimeException("The device code has expired. Please try again.")
+                } else if( message.contains('access_denied') ) {
+                    throw new RuntimeException("Access denied by user")
+                } else {
+                    throw e
+                }
+            }
+
+            Thread.sleep(intervalSeconds * 1000)
+            retryCount++
+        }
+
+        throw new RuntimeException("Authentication timed out. Please try again.")
+    }
+
+
+    protected void handleEnterpriseAuth(String apiUrl) {
+        println ""
+        ColorUtil.printColored("Please generate a Personal Access Token from your Seqera Platform instance.", "cyan bold")
+        println "You can create one at: ${ColorUtil.colorize(getWebUrlFromApiEndpoint(apiUrl) + '/tokens', 'magenta')}"
+        println ""
+
+        final pat = promptPAT()
+
+        if( !pat ) {
+            throw new AbortOperationException("Personal Access Token is required for Seqera Platform Enterprise authentication")
+        }
+
+        // Save to config
+        saveAuthToConfig(pat, apiUrl)
+        ColorUtil.printColored("Personal Access Token saved to Nextflow auth config (${getAuthFile().toString()})", "green")
+    }
+
+    private String getWebUrlFromApiEndpoint(String apiEndpoint) {
+        // Convert API endpoint to web URL
+        // e.g., https://api.cloud.seqera.io -> https://cloud.seqera.io
+        //      https://cloud.seqera.io/api -> https://cloud.seqera.io
+        return apiEndpoint.replace('://api.', '://').replace('/api', '')
```

Comment:
I would avoid this, custom can use unpredictable URI patterns

Reply:
Parking for now - will hopefully get a new API field to provide this value directly in the near-ish future.